Dalamud/Interface/Internal/Windows/Data/IDataWindowWidget.cs

```diff
@@ -0,0 +1,27 @@
+﻿namespace Dalamud.Interface.Internal.Windows;
+
+/// <summary>
+/// Class representing a date window entry.
+/// </summary>
+internal interface IDataWindowWidget
+{
+    /// <summary>
+    /// Gets the Data Kind for this data window module.
+    /// </summary>
+    DataKind DataKind { get; init; }
+    
+    /// <summary>
+    /// Gets or sets a value indicating whether this data window module is ready.
+    /// </summary>
+    bool Ready { get; protected set; }
+
+    /// <summary>
+    /// Loads the necessary data for this data window module.
+    /// </summary>
+    void Load();
```

Comment:
Can you explain your justification for a `Load` call instead of using ctors? As far as I can tell from the PR, `Load()` will be called synchronously anyways (and will block `DataWindow` construction/force reloads until everything is loaded anyways). Or, in other words, under what conditions would the "not loaded" text actually be used?

Reply:
There is a "Force Reload" button in the UI that calls "Load" again on each of the modules.
The current version of the data window uses load to get the server opcodes.
I simply shifted this behavior into the widgets themselves.

Load could be handy to address one of your other comments regarding service getters.

https://github.com/goatcorp/Dalamud/blob/master/Dalamud/Interface/Internal/Windows/DataWindow.cs#L1848-L1857

Reply:
Right, I suppose my question is more why use `Load()` over a constructor and just re-instantiate the objects when Force Reload is pressed.

Again, I'm not convinced you need to make this change, just that I want decisions like this on record lest we come back six months later and ask "why did we do it this way" without the appropriate context.

Reply:
I think there's value to being able to define a constructor that can setup some things that are more complex, and then having load be the things that might need to be reloaded.

I use this design pattern in DailyDuty, for example, I will parse lumina for specific data in the constructor, and then in the Load method, which is triggered when the user logs in, is when I load the configuration files.

Having the flexibility is worth the awkwardness in my opinion.